python/ray/data/datasource/__init__.py

```diff
@@ -24,6 +24,7 @@
     DefaultFileMetadataProvider,
 )
 from ray.data.datasource.file_meta_provider import FileMetadataProvider
+from ray.data.datasource.torch_datasource import SimpleTorchDatasource
```

Comment:
Are we okay with having this defined generically in `ray.data` as opposed to having it be an application-specific Datasource that exists in `ray.air` (`ray.ml`)? @clarkzinzow @ericl

Reply:
@matthewdeng That's a good point! My vote would be having it defined in `ray.data` for now, with the rest of our datasources. Since the I/O APIs are still exposed via `ray.data`, I'd like to keep the datasources we've defined in `ray.data` for ease-of-use in our "read --> preprocess --> train" getting started flow, where we delay interaction with the AIR package until we get to the "preprocess" stage.

I don't feel too strongly about this and haven't thought through it thoroughly, so I'm happy to hear other thoughts here.

Reply:
SGTM! The original concern was that these datasources would be not general enough to want to expose to users, but I think the documentation and having them prefixed with `Simple` should address any misunderstandings.

But on that note, should these be added to the Ray docs somewhere? I saw API docs for `Datasource` but not any of the implementations.

Reply:
>But on that note, should these be added to the Ray docs somewhere? I saw API docs for Datasource but not any of the implementations.

@matthewdeng Yep I noticed that too, left a [suggestion](https://github.com/ray-project/ray/pull/23926#issuecomment-1105534061) for how to add this datasource (and others) to the docs!